### DISCUSSION TITLE
Any forecasted support for SWI-Prolog language ?

### DISCUSSION THREAD

**Discussion by Wisermans** [helix-editor/helix#809] [General] [answered]
Included in many other editors and IDEs but i don't see it in Helix. That would be a great update knowing that [SWI-Prolog](https://www.swi-prolog.org/) is certainly the most used language for AI in the education and research sector.

**Comment by kirawi** [comment, marked as answer]
It doesn't look like there's an actively maintained tree-sitter grammar for it except [this](https://github.com/Rukiza/tree-sitter-prolog). On a side note: I thought that Python was the most used language for AI?

**Reply by Wisermans** [reply]
Thanks for your reply. I just asked a request about the tree-sitter part on the [SWI-Prolog forum ](https://swi-prolog.discourse.group/t/swi-prolog-on-helix-needing-a-tree-sitter/4515)... let's see what comes next :-)

As for Python, it is an imperative 3rd [programming generation language](https://en.wikipedia.org/wiki/Programming_language_generations) where [Prolog](https://en.wikipedia.org/wiki/Prolog) is a "rules based" 5th generation language. If you want to have fun with grammar you can also have a look at [Prolog DCG](https://www.metalevel.at/prolog/dcg) that is what i would name one more "face" of Prolog programming. Some would say that Prolog is the intelligent glue to more classical languages, the same way [IBM Watson](https://en.wikipedia.org/wiki/Watson_(computer)) got written in C++ Java and ... Prolog.

**Reply by kirawi** [reply]
From the reply in that thread, it'll probably have to wait until we support LSP syntax highlighting.

**Reply by Wisermans** [reply]
... knowing that LSP goes much further than just syntax highlighting but should be the Graal for any standardization.

**Reply by kirawi** [reply]
We already support the LSP protocol and it's straightforward to add a new LSP for a language (see [`languages.toml`](https://github.com/helix-editor/helix/blob/master/languages.toml)).

**Comment by emummel20** [comment]
Hey, any updates on this? Are you still pretty much screwed for prolog syntax support? If someone's been able to make that work, mind sharing your config? Thank you.
